test/testMocking.js

```diff
@@ -500,41 +523,105 @@ describe('Mock', () => {
       preserveResolvers: true,
     });
     const testQuery = `{
+      returnListOfInt
       returnObject{
         returnInt
         returnString
       }
     }`;
     const expected = {
+      returnListOfInt: [1, 2, 3],
       returnObject: {
-        returnInt: 12,           // from the resolver
-        returnString: 'woot!?',  // from the mock
+        returnInt: 12,           // from the resolver, see a)
+        returnString: 'woot!?',  // from the mock, see b)
       },
     };
     return graphql(jsSchema, testQuery).then((res) => {
       expect(res.data).to.deep.equal(expected);
     });
   });
 
-  it('lets you mock with functions', () => {
+  it('lets you mock and resolve non-leaf types concurrently, support promises', () => {
     const jsSchema = buildSchemaFromTypeDefinitions(shorthand);
+    const resolvers = {
+      RootQuery: {
+        returnObject: () => Promise.resolve({
+          returnInt: 12,        // a) part of a Bird, should not be masked by mock
+          // no returnString returned
+        }),
+      },
+    };
+    addResolveFunctionsToSchema(jsSchema, resolvers);
     const mockMap = {
-      Bird: () => ({ returnStringArgument: (o, a) => a.s }),
+      Bird: () => ({
+        returnInt: 3,           // see a)
+        returnString: 'woot!?', // b) another part of a Bird
+      }),
     };
-    addMockFunctionsToSchema({ schema: jsSchema, mocks: mockMap });
+    addMockFunctionsToSchema({
+      schema: jsSchema,
+      mocks: mockMap,
+      preserveResolvers: true,
+    });
     const testQuery = `{
       returnObject{
-        returnStringArgument(s: "adieu")
+        returnInt
+        returnString
       }
     }`;
     const expected = {
-      returnObject: { returnStringArgument: 'adieu' },
+      returnObject: {
+        returnInt: 12,           // from the resolver, see a)
+        returnString: 'woot!?',  // from the mock, see b)
+      },
     };
     return graphql(jsSchema, testQuery).then((res) => {
       expect(res.data).to.deep.equal(expected);
     });
   });
 
+  // it('lets you mock and resolve non-leaf types concurrently, support defineProperty', () => {
```

Comment:
Can you add a longer comment here that references the code that it would be testing (which is also commented out)